### ISSUE TITLE
[Bug]  Unable to validate certificates bis

### ISSUE THREAD

**Issue by jbperrin88** [mariadb-operator/mariadb-operator#287]
**Describe the bug**

Webhook does not validate certificate
I've already see the other case on the same subject but my update (to mariadb-operator.v0.23.1) doesn't resolve this issue

i'm using Argocd to deploy the operator on openshift and i've change my values.

The certificate seems OK from cert-manager and secret ressources

```     
 webhook:
        enabled: true
        ha:
          enabled: true
          replicas: 3
        cert:
          certManager:
            enabled: true
            issuerRef:
              group: certmanager.step.sm
              kind: StepClusterIssuer
              name: stepissuer-stepclusterissuer
            renewBefore: 8h0m0s
            duration: 168h0m0s
```
on event , I see readiness probe failed 

```    
0s          Warning   Unhealthy                pod/mariadb-operator-webhook-776cf98fcf-vjlt9                    Readiness probe failed: Get "http://10.137.0.155:8081/readyz": dial tcp 10.137.0.155:8081: connect: connection refused

0s          Warning   Unhealthy                pod/mariadb-operator-webhook-776cf98fcf-vjlt9                    Readiness probe failed: Get "http://10.137.0.155:8081/readyz": dial tcp 10.137.0.155:8081: connect: connection refused

```  

And i've exactly the same log as the other case related
```
{"level":"info","ts":1701770446.2099118,"logger":"setup","msg":"Validating certs"}
{"level":"info","ts":1701770456.2198622,"logger":"setup","msg":"Validating certs"}
{"level":"error","ts":1701770466.221084,"logger":"setup","msg":"Unable to validate certificates","error":"context deadline exceeded","stacktrace":"github.com/mariadb-operator/mariadb-operator/cmd/controller.glob..func3\n\t/app/cmd/controller/webhook.go:60\ngithub.com/spf13/cobra.(*Command).execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068\ngithub.com/spf13/cobra.(*Command).Execute\n\t/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992\ngithub.com/mariadb-operator/mariadb-operator/cmd/controller.Execute\n\t/app/cmd/controller/root.go:298\nmain.main\n\t/app/cmd/main.go:27\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:267"}
```

**Is the bug related to Galera?**
no

**Expected behaviour**
webhook takes the certificate

**Environment details**:
- Kubernetes version: openshift
- mariadb-operator version:  mariadb-operator.v0.23.1 
- Install method: helm, OLM or static manifests OLM
- Install flavour: custom

**Comment by mmontes11** [comment]
Hey there,

Thanks for reporting, could you provide the following?
- Webhook logs with `logLevel = DEBUG`
- Structure (keys) of your certificate `Secret` 

I'm guessing that, since you use a custom cert-manager issuer, the Secret may be different from what we expect here:
https://github.com/mariadb-operator/mariadb-operator/blob/4b08dfde685249f32d78c941c223f4d7c81d2c03/deploy/charts/mariadb-operator/templates/webhook-deployment.yaml#L53

Resulting in the webhook being unable to find the certificate, and therefore it doesn't become ready.

Thanks!

**Comment by mmontes11** [comment]
Related:
- https://github.com/mariadb-operator/mariadb-operator/issues/285#issuecomment-1841659548

**Comment by gg-kialo** [comment]
I'm seeing this issue too while upgrading from 0.22.0 -> 0.23.1, in my case I'm using the manifests and not the helm chart.

My debug logs are a loop of:

```
{"level":"info","ts":1702909994.9951067,"logger":"setup","msg":"Validating certs"}
{"level":"debug","ts":1702909994.995675,"logger":"setup","msg":"Error reading certificate KeyPair","error":"Error parsing PKCS1 private key: x509: failed to parse private key (use ParsePKCS8PrivateKey instead for this key format)"}
{"level":"debug","ts":1702909994.9957287,"logger":"setup","msg":"Invalid certs. Retrying...","error":"Error parsing PKCS1 private key: x509: failed to parse private key (use ParsePKCS8PrivateKey instead for this key format)"}
```

I'm not using cert-manager, but generating my own certificates with `openssl`.

**Comment by mmontes11** [comment]
Hey there @gg-kialo !

> Error parsing PKCS1 private key: x509: failed to parse private key (use ParsePKCS8PrivateKey instead for this key format

By default, openssl uses PKCS8 but our PKI module tries to parse the private key as PKCS1 so it doesn't work properly with custom certificates, it is intended to be used with certificates generated either by our cert-controller or cert-manager.  

Please, see this PR:
- https://github.com/mariadb-operator/mariadb-operator/pull/309

In order to use custom certificates, set in `--validate-cert=false` in the webhook. Something which I'm quite sure that it will fix the issue reported initially by @jbperrin88 .

Closing!

This will be released this week in `v0.0.24`, please upgrade and let me know how it goes.
